### ISSUE TITLE
Trying to get property 'name' of non-object in /attestoodle/classes/utils/db_accessor.php

### ISSUE THREAD

**Issue by Hipjea** [grp-attestoodle/moodle-tool_attestoodle#64]
Error while trying to declare a new category as formation.

**Comment by mleconte** [comment]
Hi, can you tell me the line number of the error (Trying to get property 'name' of non-object..) ?

**Comment by Hipjea** [comment]
Here is the complete error :

```
Error code: ddltablenotexist
Stack trace:

    line 665 of /lib/dml/moodle_database.php: dml_exception thrown
    line 1521 of /lib/dml/moodle_database.php: call to moodle_database->where_clause()
    line 219 of /admin/tool/attestoodle/classes/utils/db_accessor.php: call to moodle_database->get_record()
    line 141 of /admin/tool/attestoodle/classes/factories/activities_factory.php: call to tool_attestoodle\utils\db_accessor->get_course_modules_infos()
    line 56 of /admin/tool/attestoodle/classes/factories/courses_factory.php: call to tool_attestoodle\factories\activities_factory->retrieve_activities_by_course()
    line 79 of /admin/tool/attestoodle/classes/factories/courses_factory.php: call to tool_attestoodle\factories\courses_factory->create()
    line 86 of /admin/tool/attestoodle/classes/factories/trainings_factory.php: call to tool_attestoodle\factories\courses_factory->retrieve_courses_childof_category()
    line 66 of /admin/tool/attestoodle/classes/factories/trainings_factory.php: call to tool_attestoodle\factories\trainings_factory->create()
    line 58 of /admin/tool/attestoodle/index.php: call to tool_attestoodle\factories\trainings_factory->create_trainings()

Output buffer: <br /> <b>Notice</b>: Trying to get property 'name' of non-object in <b>/admin/tool/attestoodle/classes/utils/db_accessor.php</b> on line <b>208</b><br /> 
```

I managed to fix it by adding two control statements : [https://github.com/Hipjea/moodle-tool_attestoodle/commit/c78e4cd5b56d165d740b6d0478d4bcbecb079a97](https://github.com/Hipjea/moodle-tool_attestoodle/commit/c78e4cd5b56d165d740b6d0478d4bcbecb079a97)

I believe it happens because sometimes Moodle fails from to delete some modules rows when mods are uninstalled. This error happened with a non-existing module id. I checked in the database.

**Comment by mleconte** [comment]
Thank you for your contribution.
This project is being tested in several French universities until April 04, so a new version will be implemented in early April.
It will contain several optimizations, the Privacy API, **your correction** and several small evolutions.
See you next time

**Comment by mleconte** [comment]
your code is included in version 1.3.1
Sincerely
